src/servers/grpc_server_v2.cc

```diff
@@ -2419,46 +2439,69 @@ InferResponseCompleteCommon(
-        //
-        // Check if the output can be classification results
-        // (no raw_contents and not using shared memory)
-        if (output.contents().raw_contents().size() == 0) {
-          if ((alloc_payload.shm_map_ == nullptr) ||
-              (alloc_payload.shm_map_->find(output.name()) ==
-               alloc_payload.shm_map_->end())) {
-            size_t element_cnt = shape[0] * shape[1];
-            const char* base = nullptr;
-            size_t byte_size;
-            TRITONSERVER_MemoryType mem_type;
-            int64_t mem_id;
-            err = TRITONSERVER_InferenceResponseOutputData(
-                iresponse, output.name().c_str(), (const void**)&base,
-                &byte_size, &mem_type, &mem_id);
-            if (err != nullptr) {
-              break;
-            }
-            size_t offset = 0;
-            if (base != nullptr) {
-              for (size_t idx = 0; idx < element_cnt; idx++) {
-                size_t length =
-                    *(reinterpret_cast<const uint32_t*>(base + offset));
-                offset += sizeof(uint32_t);
-                output.mutable_contents()->add_byte_contents(
-                    base + offset, length);
-                offset += length;
-              }
-            }
+    // If this output was requested as classification then remove the
+    // raw output from the response and instead return classification
+    // results as a string tensor
+    const auto itr = alloc_payload.classification_map_.find(name);
+    if (itr == alloc_payload.classification_map_.end()) {
+      // Not classification...
+      output->set_datatype(TRITONSERVER_DataTypeString(datatype));
+      for (size_t idx = 0; idx < dim_count; idx++) {
+        output->add_shape(shape[idx]);
+      }
+    } else {
+      // Classification
+      const uint32_t classification_count = itr->second;
+
+      // Determine the batch1 byte size of the tensor... needed when
+      // the response tensor batch-size > 1 so that we know how to
+      // stride though the tensor data.
+      size_t batch1_element_count = 1;
+      for (size_t idx = ((batch_size == 0) ? 0 : 1); idx < dim_count; idx++) {
+        batch1_element_count *= shape[idx];
+      }
+
+      const size_t batch1_byte_size =
+          batch1_element_count * TRITONSERVER_DataTypeByteSize(datatype);
+
+      // Create the classification contents
+      std::string serialized;
+
+      size_t class_offset = 0;
+      for (uint32_t bs = 0; bs < std::max((uint32_t)1, batch_size); ++bs) {
+        std::vector<std::string> class_strs;
+        RETURN_IF_ERR(TopkClassifications(
+            iresponse, output_idx,
+            reinterpret_cast<const char*>(base) + class_offset,
+            ((class_offset + batch1_byte_size) > byte_size) ? 0
+                                                            : batch1_byte_size,
+            datatype, classification_count, &class_strs));
+
+        // Serialize for binary representation...
```

Comment:
Always return in binary form?

Reply:
For GRPC Triton always returns in raw_contents format.